pyro/infer/mcmc/mh.py

```diff
@@ -0,0 +1,255 @@
+from __future__ import absolute_import, division, print_function
+
+import pyro
+import pyro.distributions as dist
+from pyro import poutine
+from pyro.infer.mcmc.trace_kernel import TraceKernel
+from pyro.util import ng_ones, ng_zeros
+from torch import Tensor as T
+from torch.autograd import Variable as V
+from numpy.random import choice
+from numpy import isfinite
+from torch import log as tlog
+
+_R = '_RETURN'
+
+
+def VT(val):
+    return V(T(val))
+
+
+def VTA(val):
+    return VT([val])
+
+
+def assert_valid_trace(trace):
+    assert not any(['fn' in n and isinstance(n['fn'], pyro._Subsample)
```

Comment:
Prefer raising `NotImplementedError` here. (In general, we shouldn't use assertions for user-facing errors, only for programmer errors).
```py
if any(...):
   raise NotImplementedError("...")
```